tensorflow_addons/callbacks/tests/avg_model_checkpoint_test.py

```diff
@@ -0,0 +1,199 @@
+import os
+import pytest
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow_addons.callbacks import AverageModelCheckpoint
+from tensorflow_addons.optimizers import MovingAverage
+
+TRAIN_SAMPLES = 10
+NUM_CLASSES = 2
+INPUT_DIM = 3
+NUM_HIDDEN = 5
+BATCH_SIZE = 5
+EPOCHS = 5
+
+
+def get_data_and_model(optimizer="moving_avg"):
+    x = tf.random.normal([TRAIN_SAMPLES, INPUT_DIM])
+    y = tf.random.normal([TRAIN_SAMPLES, NUM_CLASSES])
+    moving_avg = MovingAverage(
+        tf.keras.optimizers.SGD(lr=2.0), sequential_update=True, average_decay=0.5
+    )
+    if optimizer == "moving_avg":
+        optimizer = moving_avg
+    inputs = keras.layers.Input(INPUT_DIM)
+    hidden_layer = keras.layers.Dense(
+        NUM_HIDDEN, input_dim=INPUT_DIM, activation="relu"
+    )(inputs)
+    outputs = keras.layers.Dense(NUM_CLASSES, activation="softmax")(hidden_layer)
+    model = keras.Model(inputs=inputs, outputs=outputs)
+    model.compile(loss="categorical_crossentropy", optimizer=optimizer, metrics=["acc"])
+    return x, y, model
+
+
+def test_compatibility_with_some_opts_only(tmpdir):
```

Comment:
Please use `tmp_path` instead of `tmpdir`, the reason is that it uses pathlib, which is a modern library to handle paths. Going forward, we're going to try to remove progressively the usage of `os.path` in addons to make the codebase more readable and bug-free.

Reply:
Yes, we should rely on `pathlib`